nzoz/src/main/java/com/example/nzoz/demonzoz/service/CityNameResolverServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.example.nzoz.demonzoz.service;
+
+import com.example.nzoz.demonzoz.dto.GoogleApiDto;
+import com.example.nzoz.demonzoz.dto.GoogleApiDto.Geometry.Location;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class CityNameResolverServiceImpl implements CityNameResolverService {
+
+  @Value("${google.maps.api.key}")
+  private String GOOGLE_MAPS_API_KEY;
+
+  private RestTemplate restTemplate;
+
+  public CityNameResolverServiceImpl(RestTemplate restTemplate) {
+    this.restTemplate = restTemplate;
+  }
+
+  @Override
+  public Location resolveCityName(final String cityName) {
+    GoogleApiDto googleApiDto = restTemplate.getForObject(
+        "https://maps.googleapis.com/maps/api/geocode/json?address=" + cityName + ",+PL&key=" + GOOGLE_MAPS_API_KEY, GoogleApiDto.class);
+
+    Location location = googleApiDto.getResults()
+        .get(0)
```

Comment:
a ty dalej... ;P

Reply:
Dodałem klucz, problem się już nie pojawia ale nie wiem co miałbym robić w takim przypadku jeśli jakimś cudem to będzie puste? Mam to łapać? I tak się bez tego request nie wykona

Reply:
dobra,olej